client/my-sites/plugins/plugins-browser/index.jsx

```diff
@@ -18,92 +18,64 @@ import NavTabs from 'components/section-nav/tabs';
 import NavItem from 'components/section-nav/item';
 import NoResults from 'my-sites/no-results';
 import PluginsBrowserList from 'my-sites/plugins/plugins-browser-list';
-import PluginsListStore from 'lib/plugins/wporg-data/list-store';
-import PluginsActions from 'lib/plugins/wporg-data/actions';
 import EmptyContent from 'components/empty-content';
 import URLSearch from 'lib/mixins/url-search';
 import infiniteScroll from 'lib/mixins/infinite-scroll';
 import JetpackManageErrorPage from 'my-sites/jetpack-manage-error-page';
 import FeatureExample from 'components/feature-example';
 import { hasTouch } from 'lib/touch-detect';
 import { recordTracksEvent } from 'state/analytics/actions';
+import { fetchPluginsList } from 'state/plugins/wporg/actions';
+import { canFetchList, getList, getShortList, isFetchingList, getCurrentSearchTerm } from 'state/plugins/wporg/selectors';
+import QueryPluginLists from 'components/data/query-plugin-lists';
+
+const SHORT_LIST_LENGTH = 6;
+const FIST_PAGE = 1;
 
 const PluginsBrowser = React.createClass( {
 
 	displayName: 'PluginsBrowser',
-	_SHORT_LIST_LENGTH: 6,
 
 	visibleCategories: [ 'new', 'popular', 'featured' ],
 
 	mixins: [ infiniteScroll( 'fetchNextPagePlugins' ), URLSearch ],
 
 	componentDidMount() {
-		PluginsListStore.on( 'change', this.refreshLists );
-		this.props.sites.on( 'change', this.refreshLists );
-
 		if ( this.props.search && this.props.searchTitle ) {
 			this.props.recordTracksEvent( 'calypso_plugins_search_noresults_recommendations_show', {
 				search_query: this.props.search
 			} );
 		}
 	},
 
-	getInitialState() {
-		return this.getPluginsLists( this.props.search );
-	},
-
-	componentWillUnmount() {
-		PluginsListStore.removeListener( 'change', this.refreshLists );
-		this.props.sites.removeListener( 'change', this.refreshLists );
-	},
-
-	componentWillReceiveProps( newProps ) {
-		this.refreshLists( newProps.search );
-	},
-
-	refreshLists( search ) {
-		this.setState( this.getPluginsLists( search || this.props.search ) );
-	},
-
-	fetchNextPagePlugins() {
-		let doSearch = true;
-
-		if ( this.state.fullLists.search && this.state.fullLists.search.fetching ) {
-			doSearch = false;
-		}
-
-		if ( this.state.fullLists.search && this.state.fullLists.search.list && this.state.fullLists.search.list.length < 10 ) {
-			doSearch = false;
-		}
-
-		if ( this.props.search && doSearch ) {
-			PluginsActions.fetchNextCategoryPage( 'search', this.props.search );
-		} else if ( this.props.category ) {
-			PluginsActions.fetchNextCategoryPage( this.props.category );
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.search ) {
+			if ( nextProps.search !== this.props.search ) {
```

Comment:
this nesting is redundant. even if `nextProps.search` is `undefined` it will successfully test for equality. also if we want to make sure we don't search for `undefined` then we can simply join the conditional

```js
if ( nextProps.search !== this.props.search ) {
	this.fetchNextPagePlugins( nextProps.search );
}
```